src/matrixroomhandler.ts

```diff
@@ -386,6 +386,139 @@ export class MatrixRoomHandler {
     return Promise.reject({err: "Unsupported", code: HTTP_UNSUPPORTED});
   }
 
+  public async HandleDiscordCommand(msg: Discord.Message) {
+    if (!msg.member.hasPermission("ADMINISTRATOR")) {
+      msg.channel.sendMessage("**ERROR:** insufficiant permissions to use matrix commands");
+      return;
+    }
+    if (!(<Discord.TextChannel> msg.channel).guild) {
+      msg.channel.sendMessage("**ERROR:** only available for guild channels");
+    }
+    const prefix = "!matrix ";
+    let command = "help";
+    let args = [];
+    if (msg.content.length >= prefix.length) {
+      const allArgs = msg.content.substring(prefix.length).split(" ");
+      if (allArgs.length && allArgs[0] !== "") {
+        command = allArgs[0];
+        allArgs.splice(0, 1);
+        args = allArgs;
+      }
+    }
+    
+    let replyMessage = "Error, unkown command. Try `!matrix help` to see all commands";
+    
+    const intent = this.bridge.getIntent();
+    const doAction = async (funcKey, action) => {
```

Comment:
I wonder if this would be better as a private function rather than embedded in a function, would make for prettier typescript

Reply:
I don't think it'd be too good to make that a private, as it is pretty specifically tailored.While it takes use of `intent` and `replyMessage` could be a return value, i have the feeling that for it being a private it should have some more abstract output maybe, not something such specific....i'm not sure